[EMCAL-842, EMCAL-912] Add more calibration objects to CalibLoader, use calib loader in reco workflow

mfasDa opened the pull request:
Added support for calibration objects:
- Temperature calibration
- Time slewing correction (not yet implemented)
- FEE DCS parameters
- BCM scale factors
Added support for CCDB-based configurable params:
- Calibration params (not yet implemented)
- Reconstruction params
- Simulation params

Use CalibLoader in reco workflow for CCDB access of
- RecoParams

Access in the reco workflow is prepared for
- FeeDCS

mfasDa committed:
[EMCAL-842] Add more calibration objects to CalibLoader

Added support for calibration objects:
- Temperature calibration
- Time slewing correction (not yet implemented)
- FEE DCS parameters
- BCM scale factors
Added support for CCDB-based configurable params:
- Calibration params (not yet implemented)
- Reconstruction params
- Simulation params

alibuild commented:
Error while checking build/O2/fullCI for 81d2493ef0db8130999e87d723a0d71edfcbb4bb at 2023-04-09 11:25:
```
## sw/BUILD/O2-full-system-test-latest/log
Detected critical problem in logfile reco_NOGPU.log
reco_NOGPU.log:[ERROR] Workflow crashed - pid 1754 (gpu-reconstruction) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[1771:BadMapCalibSpec]: [09:10:29][ERROR] Insufficient statistics: 5 entries in lowE histo, do nothing
[ERROR] Workflow crashed - pid 1754 (gpu-reconstruction) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[ERROR] Workflow crashed - pid 1846 (qc-task-ITS-ITSTrackTask) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[ERROR] Workflow crashed - pid 1827 (primary-vertexing) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[ERROR] Workflow crashed - pid 1829 (mch-cluster-transformer) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[ERROR] Workflow crashed - pid 1814 (MIDTracker) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[ERROR] Workflow crashed - pid 1856 (qc-task-TRD-Digits) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[ERROR] Workflow crashed - pid 1828 (tpc-track-interpolation) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[ERROR] SEVERE: Device gpu-reconstruction (1754) returned with 128


## sw/BUILD/o2checkcode-latest/log
--
++ echo '========== List of errors found =========='
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/fd53b567ca24f578b79d43c70c61eca4cccdebac/slc8_x86-64/o2checkcode/1.0-local13/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/11123/81d2493ef0db8130999e87d723a0d71edfcbb4bb/build_O2_fullCI/pretty.html).

mfasDa committed:
[EMCAL-912] Use CalibLoader for CCDB access

Use CalibLoader in reco workflow for CCDB access of
- RecoParams
Access in the reco workflow is prepared for
- FeeDCS

mfasDa commented:
The commit prepares the RawToCellConverter for the access of the FeeDCS params, which will be needed in order to identify inactive links during the reconstruction. At the same step the update of the RecoParam object is already moved to the CalibLoader.

alibuild commented:
Error while checking build/O2/fullCI for c280edb67b6044e1d4d1d529045fff1b835575b6 at 2023-04-10 19:20:
```
## sw/BUILD/O2-full-system-test-latest/log
Detected critical problem in logfile reco_NOGPU.log
reco_NOGPU.log:[ERROR] Workflow crashed - pid 1205 (gpu-reconstruction) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
reco_NOGPU.log:[ERROR] Workflow crashed - pid 1292 (qc-task-EMC-CellTask) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
reco_NOGPU.log-[ERROR] SEVERE: Device gpu-reconstruction (1205) returned with 128
[1222:BadMapCalibSpec]: [17:14:29][ERROR] Insufficient statistics: 1 entries in lowE histo, do nothing
[ERROR] Workflow crashed - pid 1205 (gpu-reconstruction) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[ERROR] Workflow crashed - pid 1292 (qc-task-EMC-CellTask) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[ERROR] SEVERE: Device gpu-reconstruction (1205) returned with 128


## sw/BUILD/o2checkcode-latest/log
--
++ echo '========== List of errors found =========='
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/c8994317cacc7e6ae133291b1d3bb65c46494208/slc8_x86-64/o2checkcode/1.0-local19/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/11123/c280edb67b6044e1d4d1d529045fff1b835575b6/build_O2_fullCI/pretty.html).

shahor02 commented:
local fullCI passed, merging.
